arch/arm/core/aarch32/cortex_m/fault.c

```diff
@@ -929,52 +932,48 @@ static inline z_arch_esf_t *get_esf(uint32_t msp, uint32_t psp, uint32_t exc_ret
  * @param msp MSP value immediately after the exception occurred
  * @param psp PSP value immediately after the exception occurred
  * @param exc_return EXC_RETURN value present in LR after exception entry.
- *
+ * @param callee_regs Registers that are not part of the hardware esf (r4-r11)
  */
-void z_arm_fault(uint32_t msp, uint32_t psp, uint32_t exc_return)
+void z_arm_fault(uint32_t msp, uint32_t psp, uint32_t exc_return
+#ifdef CONFIG_EXTRA_EXCEPTION_INFO
+	, struct __callee_saved_esf *callee_regs
+#endif
+)
 {
 	uint32_t reason = K_ERR_CPU_EXCEPTION;
 	int fault = SCB->ICSR & SCB_ICSR_VECTACTIVE_Msk;
 	bool recoverable, nested_exc;
-	z_arch_esf_t *esf;
+	struct __esf *hardware_esf;
 
 	/* Create a stack-ed copy of the ESF to be used during
 	 * the fault handling process.
 	 */
-	z_arch_esf_t esf_copy;
+	z_arch_esf_t esf;
```

Comment:
Why are you renaming this? It is still a stacked copy, so why not make it clear with the name?